backend/experiment/admin.py

```diff
@@ -34,14 +42,45 @@ class ExperimentAdmin(InlineActionsModelAdminMixin, admin.ModelAdmin):
     }
 
     def export(self, request, obj, parent_obj=None):
-        """Export experiment data in JSON, force download"""
-
-        response = JsonResponse(
-            obj.export_admin(), json_dumps_params={'indent': 4})
-
-        # force download attachment
-        response['Content-Disposition'] = 'attachment; filename="' + \
-            obj.slug+'.json"'
+        """Export experiment JSON data as zip archive, force download"""
+
+        # Init empty querysets
+        all_results = Result.objects.none()
+        all_songs = Song.objects.none()
+        all_sections = Section.objects.none()
+        all_participants = Participant.objects.none()
+        all_profiles = Result.objects.none()
+
+        # Collect data
+        all_sessions = obj.export_sessions().order_by('pk')
+
+        for session in all_sessions:
+            all_results |= session.export_results()
```

Comment:
I'm not familiar enough with the `|=` operator to judge: but won't we overwrite the list of `all_results` for every session, hence only export the last session's data? Or does the `|=` operator actually merge these?

Reply:
I had to search for this as wel, but `|` seems to be the way to concatenate django queries. So `|=` will concatenate the new query to the existing.